### ISSUE TITLE
React native fails with JSON.serialization issue

### ISSUE THREAD

**Issue by zhigang1992** [storybookjs/react-native#13]
**Describe the bug**
Default configuration with sb stopped working after `beta.25`

**To Reproduce**
1. `npx react-native init testSB --template react-native-template-typescript`
2. `npx -p @storybook/cli sb init --type react_native`
3. Launch the app
4. Launch the storybook server

**Expected behavior**
Should not throw error

**Screenshots**
![Simulator Screen Shot - iPhone 11 - 2020-01-02 at 10 32 25](https://user-images.githubusercontent.com/1057756/71649706-da827000-2d4b-11ea-9e44-a33fed7b582c.png)

**System:**
```
Environment Info:

  System:
    OS: macOS 10.15.2
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  Binaries:
    Node: 10.16.3 - /var/folders/xx/4pg__j5s58585b86mzz27z540000gn/T/fnm-shell-1707442/bin/node
    Yarn: 1.21.1 - ~/.yarn/bin/yarn
    npm: 6.9.0 - /var/folders/xx/4pg__j5s58585b86mzz27z540000gn/T/fnm-shell-1707442/bin/npm
  Browsers:
    Chrome: 79.0.3945.88
    Safari: 13.0.4
  npmPackages:
    @storybook/addon-actions: 5.3.0-beta.25 => 5.3.0-beta.25
    @storybook/addon-links: 5.3.0-beta.25 => 5.3.0-beta.25
    @storybook/addons: 5.3.0-beta.25 => 5.3.0-beta.25
    @storybook/react-native: 5.3.0-beta.25 => 5.3.0-beta.25
    @storybook/react-native-server: 5.3.0-beta.25 => 5.3.0-beta.25
```

**Additional context**
It still works after downgrade to `beta.25` and it stops working after `beta.26` and up.

**Comment by jenskuhrjorgensen** [comment]
Thx @doctyper - [it](https://github.com/storybookjs/react-native/issues/13#issuecomment-570794970) works for me too!

My 5 cents to folks who are not familiar with patches. To apply the suggested patch (with `npm`):
1. `npm i patch-package`
2. Add script to `package.json` to apply the patch after every install: `"postinstall": "patch-package`
3. Replace 
`hooks.currentContext = context;` 
with 
`hooks.currentContext = Object.assign({}, context, { hooks: null }); ` 
in the file
`node_modules/@storybook/addons/dist/hooks.js`
4. Run `npx patch-package @storybook/addons`
5. A file named `@storybook+addons+{VERSION}.patch` in the `patches` folder will be created and applied every time you run `npm install`

**Comment by simonepizzamiglio** [comment]
> Thx @doctyper - [it](https://github.com/storybookjs/react-native/issues/13#issuecomment-570794970) works for me too!
> 
> My 5 cents to folks who are not familiar with patches. To apply the suggested patch:
> 
> 1. Replace
>    `hooks.currentContext = context;`
>    with
>    `hooks.currentContext = Object.assign({}, context, { hooks: null }); `
>    in the file
>    `node_modules/@storybook/addons/dist/hooks.js`
> 2. Run `npx patch-package @storybook/addons`
> 3. A file named `@storybook+addons+{VERSION}.patch` in the `patches` folder will be created and applied every time you run `npm install`

@jenskuhrjorgensen you still need to install `patch-package` in your dependencies though?

**Comment by davidnuvolar** [comment]
> > Thx @doctyper - [it](https://github.com/storybookjs/react-native/issues/13#issuecomment-570794970) works for me too!
> > My 5 cents to folks who are not familiar with patches. To apply the suggested patch:
> > 
> > 1. Replace
> >    `hooks.currentContext = context;`
> >    with
> >    `hooks.currentContext = Object.assign({}, context, { hooks: null }); `
> >    in the file
> >    `node_modules/@storybook/addons/dist/hooks.js`
> > 2. Run `npx patch-package @storybook/addons`
> > 3. A file named `@storybook+addons+{VERSION}.patch` in the `patches` folder will be created and applied every time you run `npm install`
> 
> @jenskuhrjorgensen you still need to install `patch-package` in your dependencies though?

@simonepizzamiglio you should add it as a dependency and then, in your package.json scripts phase add the following:
`"scripts": {
...
    "postinstall": "patch-package",
...
}`
This way, every time you run a yarn install, the patch will be applied afterwards.

**Comment by jenskuhrjorgensen** [comment]
You are absolutely right, @simonepizzamiglio and @davidnuvolar. I've updated my previous comment - thanks!

**Comment by brotsky** [comment]
Thanks @jenskuhrjorgensen for the explanation of how to create patches! That worked perfectly for me!

**Comment by brian-perlego** [comment]
Any update on this issue ?

**Comment by Gongreg** [comment]
Hey, sadly not yet. But this is on my todo list for this week.

**Comment by ghost** [comment]
This also happened to me with `@storybook/addons@5.3.12` and the patch workaround works for now

**Comment by Gongreg** [comment]
Hey, I've created a PR to fix it in the main storybook repo.

We still are in the middle of migrating out React Native from Storybook monorepo, so I can't promise when exactly the fix will be available, but I will try to get it out as soon as I can.

I am sorry that RN Storybook was abandoned for a while. Soon RN Storybook is going to be stable (that is why we are moving it out of main monorepo) and issues like these should stop popping up.

**Comment by jonathanmachado** [comment]
Sorry I need this fix please.

Thanks

**Comment by jonathanmachado** [comment]
@Gongreg please you can check this thanks

**Comment by davidnuvolar** [comment]
> @Gongreg please you can check this thanks

Hi @jonathanmachado ,
if you need this fixed urgently, there is already a working patch in this thread, posted by jenskuhrjorgensen.

**Comment by jonathanmachado** [comment]
Thanks @davidnuvolar it's working with this fix, only one question this fix remove actions logger? because in my stories I don't see actions event loggs, Thanks again

**Comment by brian-perlego** [comment]
Storybook 5.3.14 fixed the issue for me.
Thanks @Gongreg !

**Comment by davidnuvolar** [comment]
I confirm the fix is working in 5.3.14. Thank you @Gongreg !

**Comment by Gongreg** [comment]
Great! :)
